### ISSUE TITLE
one more question about fine-tuning vs transfer-learning

### ISSUE THREAD

**Issue by shootingliu** [AlexeyAB/darknet#331]
Hello @AlexeyAB 
A quick question:
I have modified the (yolo.cfg's) last filters into 45(for 4 classes) and also the classes into 4 to train my own dataset with yolo.weights:     ./darknet detector train cfg/yolo4.data cfg/yolo.cfg yolo.weights.
It really worked. I do not know weather it means fine-tuning or not.  #245

**Comment by AlexeyAB** [comment]
Hi @shootingliu 

* If all layers changed - this is transfer learning
* If only some top layers changed - this is fine tuning

For fine tuning you can add here: https://github.com/AlexeyAB/darknet/blob/ea09a6e0b38e1ddf43ffcd81d27f0506411eb8e4/cfg/yolo.cfg#L232
this line: `stopbackward=1`

I added this in some last commit: https://github.com/AlexeyAB/darknet/commit/9ac78d8b84f6a059d2cefe22a10aa60de5b3feaf

**Comment by VanitarNordic** [comment]
@AlexeyAB 

So which of the is better? fine tuning or transfer learning?

Please add some info in the readme about this parameter, `stopbackward=1`

**Comment by AlexeyAB** [comment]
@VanitarNordic Yes, I'll add it to readme.

* **Transfer-learning** (without `stopbackward=1`)  - achieves better accuracy, but trains longer and harder (can lead to `Nan`), also without `stopbackward=1` you can train from scratch without pre-trained weights

* **Fine-tuning** (with `stopbackward=1`) - has less accuracy, but trains faster and easier, you can't fine-tune without pre-trained weights
